manifests/install.pp

```diff
@@ -13,21 +13,43 @@
   $manage_repo    = $rundeck::manage_repo
   $package_ensure = $rundeck::package_ensure
   $package_source = $rundeck::package_source
-  $rdeck_home     = $rundeck::rdeck_home
+  $user           = $rundeck::user
+  $group          = $rundeck::group
+  $manage_user    = $rundeck::manage_user
+  $manage_group   = $rundeck::manage_group
+  $user_id        = $rundeck::user_id
+  $group_id       = $rundeck::group_id
+
+  if $manage_group {
+    ensure_resource(
+      'group', $group, {
+        'ensure' => 'present',
+        'gid' => $group_id
+      }
+    )
 
-  $framework_config = deep_merge($rundeck::params::framework_config, $rundeck::framework_config)
-  $projects_dir     = $framework_config['framework.projects.dir']
-  $plugin_dir       = $framework_config['framework.libext.dir']
+    if $group != 'rundeck' {
+      group { 'rundeck':
+        ensure => absent,
+      }
+    }
+  }
 
-  $user     = $rundeck::user
-  $group    = $rundeck::group
-  $user_id  = $rundeck::user_id
-  $group_id = $rundeck::group_id
+  if $manage_user {
+    ensure_resource(
```

Comment:
Don't use ensure_resource  - use the actual resource types user and group. How many of the additional metaparameters of user and group you want to support is up to you. Now that you've got manage_user/manager_group as a boolean, if someone wants a specific metaparameter that you've decided not to support, they can always go off and manage the user/group themselves.

Reply:
Updated. See my other comment, though.